source/geod24/LocalRest.d

```diff
@@ -952,37 +955,57 @@ public final class RemoteAPI (API) : API
                         auto serialized = ArgWrapper!(Parameters!ovrld)(params)
                             .serializeToJsonString();
 
+                        MonoTime start_time = MonoTime.currTime;
                         auto command = Command(C.thisTid(), scheduler.getNextResponseId(), ovrld.mangleof, serialized);
-                        C.send(this.childTid, command);
+                        auto status = C.send(this.childTid, command);
+                        MonoTime end_time = MonoTime.currTime;
 
-                        // for the main thread, we run the "event loop" until
-                        // the request we're interested in receives a response.
-                        if (is_main_thread)
+                        if ((status == C.SendStatus.queue) || (status == C.SendStatus.success))
                         {
-                            bool terminated = false;
-                            runTask(() {
-                                while (!terminated)
-                                {
-                                    C.receiveTimeout(10.msecs,
-                                        (Response res) {
-                                            scheduler.pending = res;
-                                            scheduler.waiting[res.id].c.notify();
-                                        });
-
-                                    scheduler.yield();
-                                }
-                            });
-
-                            Response res;
-                            scheduler.start(() {
-                                res = scheduler.waitResponse(command.id, this.timeout);
-                                terminated = true;
-                            });
-                            return res;
+                            Duration timeout = this.timeout;
+                            if (this.timeout != Duration.init)
+                            {
+                                timeout = this.timeout - (end_time - start_time);
+                                if (timeout.isNegative)
```

Comment:
If I understand correctly, this part is to account for time taken by both `send` and `wait`.
This is good, and probably what the user expect (@AndrejMitrovic : Remember Swarm's timeout? ;) ).

However, if we are negative here, doesn't that mean we already timed out on `send` ?
Also there is no handling inside `send` itself: This means that if `send` has the ability to block indefinitely, the timeout feature won't work. If it does not have the ability to block indefinitely, then is including it in the timeout really worth it ? As long as it is bounded within a certain value, I don't think it matters as much.

On a similar topic, there's two school of thought about async IO: If you set `O_NONBLOCK` on a normal file, it will never actually return `E_WOULDBLOCK` because the disk can be _slow_, but not block _indefinitely_. However, on empty sockets or pipes, it would because it could block _indefinitely_. Now some people started to be really annoyed by this behavior, as reading from the disk can be _really_ slow.
We have a similar problem here: If `send` is fast and non-blocking, there's no point including it in the timeout. If it can block indefinitely, it *needs* to be included in the timeout. If it cannot block indefinitely but can be arbitrarily slow, we might want to.